wa/framework/workload.py

```diff
@@ -218,7 +229,25 @@ def teardown(self, context):
 
     @once_per_instance
     def finalize(self, context):
-        pass
+        self.remove_assets(context)
+
+    def deploy_assets(self, context, target_directory=None):
+        if not self.deployable_assets:
+            return
+
+        if not target_directory:
+            target_directory = self.target.working_directory
+        self.target.execute('mkdir -p {}'.format(target_directory))
```

Comment:
Why is this necessary? Shouldn't the working_directory already exist by the time we get to initializing workloads?

Reply:
The idea here was that you could set a custom directory for the assets to be deployed to instead of the working directory so the custom folder may not have existed. However I've re-done this bit to be part of the base Workload Class and to use an attribute to store the directory path instead.